File: app/src/main/java/com/example/point/ShowActivity.java
package com.example.point;

import android.content.ComponentName;
import android.content.Context;
import android.content.Intent;
import android.content.ServiceConnection;
import android.os.Bundle;
import android.os.IBinder;
import android.util.Log;

import androidx.annotation.Nullable;
import androidx.appcompat.app.AppCompatActivity;

/**
 * @author YangJ
 */
public class ShowActivity extends AppCompatActivity {

    private ServiceConnection mConnection;

    // TAG
    private static final String TAG = "ShowActivity";

    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_show);
        initService();
    }

    @Override
    protected void onDestroy() {
        super.onDestroy();
        unbindService(mConnection);
    }

    private void initService() {
        Intent intent = new Intent(this, MyService.class);
        mConnection = new ServiceConnection() {
            @Override
            public void onServiceConnected(ComponentName name, IBinder service) {
                Log.i(TAG, "onServiceConnected: " + name);
            }

            @Override
            public void onServiceDisconnected(ComponentName name) {
                Log.i(TAG, "onServiceDisconnected: " + name);
            }
        };
        bindService(intent, mConnection, Context.BIND_AUTO_CREATE);
    }

}

File: app/src/main/java/com/example/point/ShowActivity.kt
package com.example.point

import android.content.ComponentName
import android.content.Context
import android.content.Intent
import android.content.ServiceConnection
import android.os.Bundle
import android.os.IBinder

import androidx.appcompat.app.AppCompatActivity

/**
 * @author YangJ
 */
class ShowActivity : AppCompatActivity() {

    private lateinit var mConnection: ServiceConnection

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_show)
        initService()
    }

    private fun initService() {
        val intent = Intent(this, MyService::class.java)
        mConnection = object : ServiceConnection {
            override fun onServiceDisconnected(name: ComponentName?) {
                println("onServiceDisconnected: $name")
            }

            override fun onServiceConnected(name: ComponentName?, service: IBinder?) {
                println("onServiceConnected: $name")
            }

        }
        bindService(intent, mConnection, Context.BIND_AUTO_CREATE)
    }

    override fun onDestroy() {
        super.onDestroy()
        unbindService(mConnection)
    }
}
